js/apps/account-ui/src/oid4vci/Oid4Vci.tsx

```diff
@@ -0,0 +1,204 @@
+import {
+    Select,
+    SelectList,
+    SelectOption,
+    PageSectionVariants,
+    PageSection,
+    ActionList,
+    ActionListItem,
+    List,
+    ListItem,
+    MenuToggleElement,
+    MenuToggle
+} from '@patternfly/react-core';
+import { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useAlerts, useEnvironment } from "@keycloak/keycloak-ui-shared";
+import { usePromise } from "../utils/usePromise";
+import { Page } from "../components/page/Page";
+import { parseResponse } from "../api/parse-response";
+import { token } from '../api/request';
+
+type CredentialsIssuer = {
+  credential_issuer: string;
+  credential_endpoint: string;
+  authorization_servers: string[]; 
+  credential_configurations_supported: Map<string, SupportedCredentialConfiguration>
+}
+
+type SupportedCredentialConfiguration = {
+  id: string,
+  format: string,
+  scope: string
+}
+
+type VCState = {
+  dropdownItems: string[],
+  selectOptions: Map<string, SupportedCredentialConfiguration>,
+  credentialIssuer?: CredentialsIssuer,
+  issuerDid: string,
+  qrCode: string,
+  isOpen: boolean,
+  offerQRVisible: boolean
+}
+
+export const Oid4Vci = () => {
+    const { addAlert, addError } = useAlerts();
+    const context = useEnvironment();
+   
+    const { t } = useTranslation();
+    const initialState: VCState = {
+      dropdownItems: [],
+      selectOptions: new Map<string, SupportedCredentialConfiguration>(),
+      issuerDid: "",
+      qrCode: "",
+      isOpen: false,
+      offerQRVisible: false
+    }
+    const initialSelected = t('verifiableCredentialsSelectionDefault') 
+
+    const [selected, setSelected] = useState<string>(initialSelected);
+    const [vcState, setState] = useState<VCState>(initialState);
+    const url:string = context.environment.authUrl
+    const realm = context.environment.realm
+    const wellKnownIssuer = url + "/realms/" + realm + "/.well-known/openid-credential-issuer"
+
+    usePromise(
+      (signal) =>
+          getIssuer(wellKnownIssuer),
+      (issuer) => {
+        const ccsMap = new Map(Object.entries(issuer.credential_configurations_supported))
+        const ccsKeyArray = Array.from(ccsMap.keys());
+        setState({...vcState , credentialIssuer: issuer, dropdownItems: ccsKeyArray, selectOptions: ccsMap});
```

Comment:
Since the `dropDownItems` are derived from other state, you should instead save the fetched `issuer` in a separate state and use [`useMemo()` ](https://react.dev/reference/react/useMemo) to compute the derived value for `dropdownItems`, rather than keep a dedicated state that needs to be synchronized manually.

Reply:
Fixed in https://github.com/keycloak/keycloak/pull/29174/commits/26857215ec6f6053560781e6ce9849202a601bee